docs/rtk-query/usage-with-typescript.mdx

```diff
@@ -497,3 +497,184 @@ function MaybePost({ id }: { id?: number }) {
   return <div>...</div>
 }
 ```
+
+## Type safe error handling
+
+When an error is gracefully provided from a [`base query`](./api/createApi.mdx#baseQuery), RTK query will provide the error
+directly. If an unexpected error is thrown by user code rather than a graceful handled error,
+that error will be transformed into a `SerializedError` shape. Users should ensure that they are checking which
+kind of error they are dealing with prior to attempting to access it's properties. This can be done in a type safe manner either
```

Comment:
```suggestion
that error will be transformed into a `SerializedError` shape. Users should make sure that they are checking which kind of error they are dealing with before attempting to access its properties. This can be done in a type safe manner either
```